src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -342,18 +374,16 @@ def _check_registry_health(cmd, registry_name, ignore_errors):
                 _handle_error(CMK_MANAGED_IDENTITY_ERROR.format_error_message(registry_name), ignore_errors)
 
 
-def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # pylint: disable=too-many-locals, too-many-statements
+def _check_private_endpoint(cmd,
+                            registry_name,
+                            registry,
```

Comment:
why do you need registry and registry name?

Reply:
'Registry' is an object passed down from 'get_registry_by_name() on line 548.  The other methods like '_check_private_endpoint' were using 'get_registry_by_name()" and accessing properties from 'registry' so I changed it so that 'get_registry..." is called once and then passes the registry properties down.

 'Registry_name' is the name of the registry itself.

![image](https://github.com/Azure/azure-cli/assets/82487617/015a4682-a534-45ed-b71c-af00514e8954)

Reply:
I do like the optimization however, can't we remove the registry name argument then. 
Or are you making it optional (either one of the 2)?

Reply:
Yes, you are correct, I can just use the 'registry' object and the registry name is accessible already within that; will change this, thank you.